File: JAVA/ml/ML_Campaigns.java
package ml;

import java.util.Map;

import org.json.JSONObject;

import base.ML_Rest;

/**
 * Managing Campaigns class for MailerSoft.
 * 
 * @author Vaidotas Strazdas
 *
 */
public class ML_Campaigns extends ML_Rest {

	/**
	 * Constructor.
	 * 
	 * @param apiKey API Key
	 */
	public ML_Campaigns(String apiKey) {
		super(apiKey);
		this.name = "campaigns";
		this.setPath(null);
	}
	
	/**
	 * Overloaded ML_Rest setId() method, so programmer
	 * would not need any casting.
	 */
	public ML_Campaigns setId(long id) {
		super.setId(id);
		return this;
	}
	
	/**
	 * Get recipients.
	 * 
	 * @param data Specific data about recipients.
	 * @return Recipients list.
	 * @throws Exception
	 * @see JSONObject
	 */
    public JSONObject getRecipients(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "recipients");
    	return result;
    }
	
    /**
     * Overloaded getRecipients() method.
     * 
     * @return Recipients list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getRecipients() throws Exception {
    	JSONObject result = this.getRecipients(null);
    	return result;
    }
    
    /**
     * Get opens.
     * 
     * @param data Specific data about opens.
     * @return Opens list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getOpens(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "opens");
    	return result;
    }

    /**
     * Overloaded getOpens() method.
     * 
     * @return Opens list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getOpens() throws Exception {
    	JSONObject result = this.getOpens(null);
    	return result;
    }

    /**
     * Get clicks.
     * 
     * @param data Specific data about clicks.
     * @return Clicks list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getClicks(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "clicks");
    	return result;
    }

    /**
     * Overloaded getClicks() method.
     * 
     * @return Clicks list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getClicks() throws Exception {
    	JSONObject result = this.getClicks(null);
    	return result;
    }
    
    /**
     * Get unsubscribed users.
     * 
     * @param data Specific data about unsubscribed users list.
     * @return Unsubscribed users list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getUnsubscribes(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "unsubscribes");
    	return result;
    }

    /**
     * Overloaded getUnsubscribes() method.
     * 
     * @return Unsubscribed users list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getUnsubscribes() throws Exception {
    	JSONObject result = this.getUnsubscribes(null);
    	return result;
    }
    
    /**
     * Get bounces.
     * 
     * @param data Specific data about bounces.
     * @return Bounces list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getBounces(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "bounces");
    	return result;
    }

    /**
     * Overloaded getBounces() method.
     * 
     * @return Bounces list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getBounces() throws Exception {
    	JSONObject result = this.getBounces(null);
    	return result;
    }
    
    /**
     * Get users who complained about junk.
     * 
     * @param data Specific data about users.
     * @return Users complaining about junk list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getJunk(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "junk");
    	return result;
    }

    /**
     * Overloaded getJunk() method.
     * 
     * @return Users complaining about junk list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getJunk() throws Exception {
    	JSONObject result = this.getJunk(null);
    	return result;
    }
    
}

File: Python/ML_Campaigns.py
from ML_Rest import ML_Rest

class ML_Campaigns(ML_Rest):
    """
    ML_Campaigns class used to manage Campaigns.
    """
    def __init__(self, api_key):
        """
        Constructor for the ML_Campaigns class.
        
        :param api_key: Your API Key for MailerSoft.
        :type api_key: str
        """
        self._mc_parent = ML_Rest
        self._mc_parent.__init__(self, api_key)
        self._name = "campaigns"
        self._set_path()
    
    def get_recipients(self, data={}):
        """
        Get recipients from MailerSoft for a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'recipients')
    
    def get_opens(self, data={}):
        """
        Retrieve a paged result representing all the subscribers that opened a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients that opened a given campaign.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'opens')
    
    def get_clicks(self, data={}):
        """
        Retrieve a paged result representing all the subscribers that clicked on a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients that clicked a given campaign.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'clicks')
    
    def get_unsubscribes(self, data={}):
        """
        Retrieve a paged result representing all the subscribers that unsubscribed from a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients that unsubscribed from a given campaign.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'unsubscribes')
    
    def get_bounces(self, data={}):
        """
        Retrieve a paged result representing all the subscribers who bounced for a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients who bounced for a given campaign.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'bounces')
    
    def get_junk(self, data={}):
        """
        Retrieve a paged result representing all the subscribers who complained about a given campaign.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with recipients who complained about a given campaign.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, 'junk')